doc/reference/guidelines.rst

```diff
@@ -0,0 +1,48 @@
+.. highlight:: python
+
+=================
+Coding Guidelines
+=================
+
+Python
+======
+
+- PEP8 ignore E501
+- prefer % over format
+- try to avoid generators and decorators
```

Comment:
Seriously?  It seems completely arbitrary.  Why would we want to exclude generators?  Do we really have a problem with generators in our python code?  If so, I suspect we actually have a problem with bad coders doing weird stuff, but then ban weird stuff instead of generators.